### PR TITLE
Add dnsmasq DHCP to ci_network and devscripts

### PR DESCRIPTION
**Pull request by pablintino**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: roles/ci_network/tasks/apply-dnsmasq.yml [hunk 1]
```diff
@@ -36,46 +36,73 @@
   become: true
   when: >
     ("'127.0.0.1' not in ansible_dns.nameservers") or
-    ("'forwarders' in cifmw_network_local_dns")
+    ("'forwarders' in cifmw_network_dnsmasq_config")
```
Comment (cjeanner) [inline roles/ci_network/tasks/apply-dnsmasq.yml:39]: care to mention that parameter in the defaults/main.yml? Even as a comment - that would be nice for ppl looking "only" at that file.

Comment (pablintino) [reply roles/ci_network/tasks/apply-dnsmasq.yml:39]: Done.

File: roles/ci_network/README.md [hunk 2]
```diff
@@ -68,3 +74,21 @@ cifmw_network_local_dns:
   forwarders:
     - 8.8.8.8         # Optional. list, of DNS forwarders to be applied.
 ```
+
+
+## DHCP configuration
+
+To enable dnsma configuration is represented by
```
Comment (cjeanner) [inline roles/ci_network/README.md:81]: typo, raised in spelling.

**Comment by cjeanner** [comment]
/lgtm

at least it does break fully virtualized HCI deployment with the current scenario files. The logic and code seems good, but I can't test any further with my current lab, unfortunately.

**Review by psathyan — APPROVED** [review]

File: roles/ci_network/tasks/add-static-leases.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+---
+- name: Add DHCP reservations
+  when:
+    - cifmw_network_dnsmasq_config is defined
+    - "'dhcp' in cifmw_network_dnsmasq_config"
+    - "'network' in cifmw_network_dnsmasq_config['dhcp']"
```
Comment (rebtoor) [inline roles/ci_network/tasks/add-static-leases.yml:6]: why the double quote here?

Comment (cjeanner) [reply roles/ci_network/tasks/add-static-leases.yml:6]: because of the `when foo in bar` - ansible `when` is slightly stupid on that. Unless it changed?

**Review by rebtoor — APPROVED** [review]

**Pull request merged**